internal/ethapi/api.go

```diff
@@ -1667,12 +1672,17 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 		fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
 	} else {
 		// var gasPrice *big.Int = new(big.Int)
-		if tx.Type() == types.DynamicFeeTxType || tx.Type() == types.CeloDynamicFeeTxType || tx.Type() == types.CeloDynamicFeeTxV2Type {
+		if tx.Type() == types.DynamicFeeTxType || tx.Type() == types.CeloDynamicFeeTxType || tx.Type() == types.CeloDynamicFeeTxV2Type || tx.Type() == types.CeloDenominatedTxType {
 			header, err := s.b.HeaderByHash(ctx, blockHash)
 			if err != nil {
 				return nil, err
 			}
-			gasPriceMinimum, err := s.b.GasPriceMinimumForHeader(ctx, tx.FeeCurrency(), header)
+			var currency *common.Address = tx.FeeCurrency()
+			// Celo Denominated txs are denominated in celo
+			if tx.Type() == types.CeloDenominatedTxType {
+				currency = nil
+			}
+			gasPriceMinimum, err := s.b.GasPriceMinimumForHeader(ctx, currency, header)
```

Comment:
Do we need this change at all? It looks like running into the `else` branch would be fine.

Reply:
I'm not sure, because the else is crafted for the legacytx that are not eip1559 and don't have a tip.
but i've introduced a DenominatedFeeCurrency in some commits after this one, that I think can be used here and it will look simpler.